docs/poly-to-normal-variant/index.md

```diff
@@ -0,0 +1,345 @@
+# From Polymorphic to Normal Variant
+
+The `Promo` component works fine now, but let's take a moment to explore how to
+implement it with a normal variant instead of a polymorphic variant. This small
+refactor should give us more insight into OCaml's type system. Additionally,
+normal variants are better than polymorphic variants at "documenting" the types
+that will be used in your program, since they must always be explicitly defined
```

Comment:
It's nice to mention documentation, but this doesn't explain much about the technical use cases where normal variants are more useful than poly variants:
- Errors and signatures are simpler
- Stronger type guarantees (pattern match "completeness", safer refactoring)
- Potentially better performance (both at build time and runtime)

The downsides are that:
- Normal variants can be "expanded" dynamically, so for cases where it's unclear if more cases might be needed (or where the author of the types don't control their usage, like library authors) polyvars might fit better because extra flexibility
- They always need to be namespaced with module or annotated explicitly when used, while polyvars values can just be written without any annotations

Do you think it's worth mentioning these aspects?

Reply:
> Errors and signatures are simpler

Here, do you mean that error messages for normal variants are simpler?

Reply:
> Here, do you mean that error messages for normal variants are simpler?

Yes.

> Do you mean that normal variants _cannot_ be expanded dynamically?

Yes, I said it the other way around 🙃